kiva/fonttools/app_font.py

```diff
@@ -31,9 +29,11 @@ def add_application_fonts(filenames):
     fm.update_fonts(filenames)
 
     # Handle the GUI toolkit
-    if ETSConfig.toolkit.startswith("qt"):
+    from traitsui.api import toolkit
+
+    if toolkit().toolkit.startswith("qt"):
         _qt_impl(filenames)
-    elif ETSConfig.toolkit == "wx":
+    elif toolkit().toolkit == "wx":
         _wx_impl(filenames)
```

Comment:
What if this code is running headless with no GUI library installed?

Reply:
You should get the null toolkit back and so nothing will happen.

The failure mode this is guarding against is if `add_application_fonts` gets called before toolkits are determined in the app.  However this will fail if TraitsUI isn't at least installed.  Probably need to guard the TraitsUI import and just do nothing if TraitsUI isn't available.  And possibly use the Pyface toolkit instead.

Reply:
Null toolkit... That's great! Since using `pyface` would require fewer dependencies for a headless project, that might be a better direction to take. That said, I don't know if there are any drawbacks to preferring `pyface` to `traitsui`.